ProjFS.Mac/PrjFSKextTests/HandleOperationTests.mm

```diff
@@ -1196,4 +1271,108 @@ - (void) testIneligibleFilesystemType
     XCTAssertFalse(MockCalls::DidCallFunction(ProviderMessaging_TrySendRequestAndWaitForResponse));
 }
 
+- (void) testOfflineRootDeniesAccessToEmptyFile
```

Comment:
Would it be possible to add a test to validate the offline rename scenarios on pre/post Mojave?

Reply:
Good idea, I've added a new test `testOfflineRootDeniesRename` as there's a bit more to do in that case and extending `testOfflineRootDeniesAccessToEmptyFile` would be messy.